amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/validators/UUIDValidator.java

```diff
@@ -22,13 +22,24 @@
     public ApiErrorMessage getErrorMessage() {
         return ActivityErrors.FIELD_INVALID_VALUE;
     }
-
+    
     @Override
     public boolean isValid(ObjectImporter importer, Map<String, Object> newFieldParent,
+                           APIField fieldDescription, String fieldPath) {
+        return isValid(importer, null, newFieldParent, fieldDescription, fieldPath);
+    }
+
+    @Override
+    public boolean isValid(ObjectImporter importer, Object currentObject, Map<String, Object> newFieldParent,
             APIField fieldDescription, String fieldPath) {
         
         if (fieldDescription.getFieldName().equals(FieldMap.underscorify(ActivityFieldsConstants.UUID))) {
-            String uuid = StringUtils.trim((String) newFieldParent.get(fieldDescription.getFieldName()));
+            String uuid = null;
+            if (!newFieldParent.containsKey(fieldDescription.getFieldName()) && importer.isUpdate()) {
```

Comment:
Looks like it doesn't matter if we update or not. While we don't have this situation but even during insert a field could have some default value that is not null.